service/test/agama/cmdline_args_test.rb

```diff
@@ -29,8 +29,27 @@
   describe ".read_from" do
     it "reads the kernel command line options and return a CmdlineArgs object" do
       args = described_class.read_from(File.join(workdir, "/proc/cmdline"))
-      expect(args.data).to eql("web" => { "ssl" => true })
+      expect(args.data["auto"]).to eq("http://mydomain.org/tumbleweed.jsonnet")
+    end
+
+    it "sets #config_url if specified on cmdline" do
+      args = described_class.read_from(File.join(workdir, "/proc/cmdline"))
       expect(args.config_url).to eql("http://example.org/agama.yaml")
     end
+
+    it "converts 'true' and  'false' values into booleans" do
+      args = described_class.read_from(File.join(workdir, "/proc/cmdline"))
+      expect(args.data["web"]).to eql({ "ssl" => true })
+    end
+
+    it "converts keys that contain another dots after 'agama.' to hash value and using first value as key" do
+      args = described_class.read_from(File.join(workdir, "/proc/cmdline"))
+      expect(args.data["web"]).to eq({ "ssl" => true })
+    end
```

Comment:
These examples has identical expectations. Is it ok?

Reply:
yes, as both test uses original fixture, so I would like to here emphasize that it do "hashisize" of key and also normalize of value. My plan was to use separate examples with documented ones from https://agama-project.github.io/docs/user/boot_options sadly noone of those options are documented, so I just keep it for testing regressions.